pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -549,18 +549,13 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 		Name:     nicName,
 		VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
 	}
-	switch set := s.scope.Machine.ObjectMeta.Labels[v1alpha1.MachineRoleLabel]; set {
```

Comment:
If the Subnet is empty you could populate the hardcoded ones above just for backward compat.

Something like

if subnet == "" {
} else {
  //fallback
}

Eventually the fallback code can be removed